conda/installers/conda.py

```diff
@@ -0,0 +1,57 @@
+# Copyright (C) 2012 Anaconda, Inc
+# SPDX-License-Identifier: BSD-3-Clause
+"""Conda-flavored installer."""
+
+import tempfile
+from typing import Iterable
+from os.path import basename
+
+from boltons.setutils import IndexedSet
+
+from ..base.constants import UpdateModifier
+from ..base.context import Context
+from ..common.constants import NULL
+from ..exceptions import UnsatisfiableError
+from ..models.channel import Channel, prioritize_channels
+
+
+def _solve(prefix, specs, context: Context):
+    """Solve the environment"""
+    # TODO: support all various ways this happens
+    # Including 'nodefaults' in the channels list disables the defaults
+    channel_urls = context.channels
+    _channel_priority_map = prioritize_channels(channel_urls)
+
+    channels = IndexedSet(Channel(url) for url in _channel_priority_map)
+    subdirs = IndexedSet(basename(url) for url in _channel_priority_map)
+
+    solver_backend = context.plugin_manager.get_cached_solver_backend()
+    solver = solver_backend(prefix, channels, subdirs, specs_to_add=specs)
+    return solver
+
+
+def dry_run(specs: Iterable[str], context: Context, *args, **kwargs) -> Iterable[str]:
+    """Do a dry run of the environment solve"""
+    solver = _solve(tempfile.mkdtemp(), specs, context)
+    pkgs = solver.solve_final_state()
+    return [str(p) for p in pkgs]
+
+
+def install(prefix: str, specs: Iterable[str], context: Context, *args, **kwargs) -> Iterable[str]:
+    """Install packages into an environment"""
```

Comment:
This still needs work in order to support all the install options required for `conda.cli.install.install`